lib/api/api_auth.go

```diff
@@ -82,14 +106,36 @@ func basicAuthAndSessionMiddleware(cookieName string, guiCfg config.GUIConfigura
 			}
 		}
 
+		// Fall back to Basic auth if provided, but don't prompt for it
 		if username, ok := attemptBasicAuth(r, guiCfg, ldapCfg, evLogger); ok {
 			createSession(cookieName, username, guiCfg, evLogger, w, r)
 			next.ServeHTTP(w, r)
+		}
+
+		forbidden(w)
+	})
+
+	handlePasswordLogin := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		var req struct{Username string; Password string}
+		err := unmarshalTo(r.Body, &req)
+		if err != nil {
+			l.Debugln("Failed to parse username and password:", err)
+			http.Error(w, "Failed to parse username and password.", 400)
 			return
 		}
 
-		unauthorized(w)
+		if auth(req.Username, req.Password, guiCfg, ldapCfg) {
+			createSession(cookieName, req.Username, guiCfg, evLogger, w, r)
+			w.WriteHeader(http.StatusNoContent)
+			return
+		}
+
+		emitLoginAttempt(false, req.Username, r.RemoteAddr, evLogger)
+		forbidden(w)
+		return
```

Comment:
Unnecessary return.

Reply:
Yep, thanks!

Reply:
Fixed in 21a8b5d3d6050a55ed60697116ea3021e53ad7a2.